engine/execution/state/state.go

```diff
@@ -219,6 +231,9 @@ func LedgerGetRegister(ldg ledger.Ledger, commitment flow.StateCommitment) delta
 			return nil, nil
 		}
 
+		// don't cache value with len zero
+		readCache[regID] = flow.RegisterEntry{Key: regID, Value: values[0]}
```

Comment:
Maybe refactor `values[0]` here and below into a common variable, to ensure the same value that's returned is also the one that is cached